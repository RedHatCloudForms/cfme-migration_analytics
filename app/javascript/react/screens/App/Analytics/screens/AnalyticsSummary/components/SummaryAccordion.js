import React from 'react';
import PropTypes from 'prop-types';
import { ListView, Grid } from 'patternfly-react';
import SummaryTable from './SummaryTable';

const SummaryAccordion = ({ summaryData }) => {
  return (
    <div className="reports-summary">
      <p>{__('CloudForms discovered the following information:')}</p>
      <div className="environment-summary">
        <h2>{__('Environment Summary')}</h2>
        <SummaryTable data={summaryData.total} />
      </div>
      <ListView>
        {summaryData.providers.map(provider => (
          <ListView.Item
            key={provider.id}
            leftContent={<ListView.Icon type="pf" name="server-group" />}
            heading={provider.name}
            description={__('Provider Summary')}
            stacked
          >
            <Grid.Row>
              <Grid.Col sm={11}>
                <SummaryTable data={provider} />
              </Grid.Col>
            </Grid.Row>
          </ListView.Item>
        ))}
      </ListView>
    </div>
  );
};

SummaryAccordion.propTypes = {
  summaryData: PropTypes.shape({
    total: PropTypes.shape({
      numProviders: PropTypes.number,
      numHypervisors: PropTypes.number,
      numVms: PropTypes.number,
      allocatedDiskSpace: PropTypes.number,
      allocatedMemory: PropTypes.number,
      numCpuCores: PropTypes.number
    }),
    providers: PropTypes.arrayOf(
      PropTypes.shape({
        numHypervisors: PropTypes.number,
        numVms: PropTypes.number,
        allocatedDiskSpace: PropTypes.number,
        allocatedMemory: PropTypes.number,
        numCpuCores: PropTypes.number
      })
    )
  })
};

export default SummaryAccordion;
