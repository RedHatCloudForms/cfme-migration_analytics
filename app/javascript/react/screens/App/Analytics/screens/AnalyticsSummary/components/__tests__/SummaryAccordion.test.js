import React from 'react';
import { shallow } from 'enzyme';
import SummaryAccordion from '../SummaryAccordion';

describe('summary accordion', () => {
  const getBaseProps = () => ({
    summaryData: {
      total: { mock: 'data' },
      providers: [{ mock: 'data' }, { mock: 'data' }]
    }
  });

  test('renders correctly', () => {
    const component = shallow(<SummaryAccordion {...getBaseProps()} />);
    expect(component).toMatchSnapshot();
  });
});
