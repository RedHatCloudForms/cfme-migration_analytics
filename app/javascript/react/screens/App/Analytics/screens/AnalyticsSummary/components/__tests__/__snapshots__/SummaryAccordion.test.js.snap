// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary accordion renders correctly 1`] = `
<div
  className="reports-summary"
>
  <p>
    CloudForms discovered the following information:
  </p>
  <div
    className="environment-summary"
  >
    <h2>
      Environment Summary
    </h2>
    <SummaryTable
      data={
        Object {
          "mock": "data",
        }
      }
    />
  </div>
  <ListView
    className=""
  >
    <ListViewItem
      actions={null}
      additionalInfo={null}
      checkboxInput={null}
      compoundExpand={false}
      compoundExpanded={false}
      description="Provider Summary"
      heading={null}
      hideCloseIcon={false}
      initExpanded={false}
      leftContent={
        <ListViewIcon
          className=""
          name="server-group"
          size="sm"
          type="pf"
        />
      }
      onCloseCompoundExpand={[Function]}
      onExpand={[Function]}
      onExpandClose={[Function]}
      stacked={true}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          sm={11}
        >
          <SummaryTable
            data={
              Object {
                "mock": "data",
              }
            }
          />
        </Col>
      </Row>
    </ListViewItem>
    <ListViewItem
      actions={null}
      additionalInfo={null}
      checkboxInput={null}
      compoundExpand={false}
      compoundExpanded={false}
      description="Provider Summary"
      heading={null}
      hideCloseIcon={false}
      initExpanded={false}
      leftContent={
        <ListViewIcon
          className=""
          name="server-group"
          size="sm"
          type="pf"
        />
      }
      onCloseCompoundExpand={[Function]}
      onExpand={[Function]}
      onExpandClose={[Function]}
      stacked={true}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          sm={11}
        >
          <SummaryTable
            data={
              Object {
                "mock": "data",
              }
            }
          />
        </Col>
      </Row>
    </ListViewItem>
  </ListView>
</div>
`;
